### PR TITLE
File dropzone docs

### PR DESCRIPTION
**Pull request by Fercas123**

### CODE REVIEW

**Review by joshwooding — DISMISSED** [review]
This PR will need to be updated to match the new content guidelines, could you take a look at https://github.com/jpmorganchase/salt-ds/pull/2787

You can see how the other docs have been updated too

File: site/docs/components/file-drop-zone/examples.mdx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
+<LivePreviewControls>
+  <LivePreview componentName="file-drop-zone" exampleName="Default">
+
+    ### FileDropZone
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/examples.mdx:15]: This title should match the example name or the display name

File: site/docs/components/file-drop-zone/examples.mdx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    This component allows users to upload files they've selected from the native file browser interface or dragged-and-dropped — with the files automatically uploaded to your web application.
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/examples.mdx:17]:
```suggestion
    This component allows users to upload files they've selected from the native file browser interface or dragged-and-dropped—with the files automatically uploaded to your web application.
```

File: site/docs/components/file-drop-zone/accessibility.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+title:
+  $ref: ./#/title
+layout: DetailComponent
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
+### Best practices
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/accessibility.mdx:12]: These best practices don't seem to be relevant to the component's accessibility. Are these better placed in usage?

File: site/docs/components/file-drop-zone/accessibility.mdx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+- Consider using the File Drop Zone within forms when it's necessary for users to attach supporting files.
+- Make sure any limits, such as maximum file size and type restrictions, are communicated to users prior to upload. If an upload failure results, ensure that the reason for the error — and how to resolve it — are clearly explained.
+- Inform users when files successfully upload to reassure them that the action has been completed.
+
+The following keyboard interactions are provided:
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/accessibility.mdx:18]:
```suggestion
### Keyboard interactions
```

File: site/docs/components/file-drop-zone/examples.mdx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    ### Success
+
+    Inform users when files successfully upload. Success messages should reassure users that the action has been completed.
+    
+  </LivePreview>
+  <LivePreview componentName="file-drop-zone" exampleName="Disabled">
+
+    ### Disabled
+    
+    The File Drop Zone component can be disabled, with no resultant action when users drop a file onto the component.
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/examples.mdx:42]: You should add guidance that mentions to make sure disabled is passed to the trigger too.

File: site/docs/components/file-drop-zone/usage.mdx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+title:
+  $ref: ./#/title
+layout: DetailComponent
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
+### Using the FileDropZone component
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/usage.mdx:12]:
```suggestion
### Using the component
```

File: site/src/examples/file-drop-zone/Default.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { ReactElement } from "react";
+import {
+  FileDropZone,
+  FileDropZoneIcon,
+  FileDropZoneTrigger
+} from "@salt-ds/lab";
+
+export const Default = (): ReactElement => {
+
+  return <FileDropZone style={{width: 200}}>
```
Comment (joshwooding) [inline site/src/examples/file-drop-zone/Default.tsx:10]: Super minor, but the formatting seems quite strange. Can you try changing this to a short-hand arrow function e.g.

```tsx
export const Default = (): ReactElement => (
	<div />
);
```

File: site/docs/components/file-drop-zone/index.mdx [hunk 8]
```diff
@@ -0,0 +1,22 @@
+---
+title: File Drop Zone
+data:
+  description: FileDropZone provides a target area for users to drag and drop files, such as documents or images, and automatically uploads them to the web application.
```
[the remark this answers is not in the document]
Comment (Fercas123) [reply site/docs/components/file-drop-zone/index.mdx:4]: they are not technically part of the component, the banner will be documented in the pattern, and the button is part of it but it can be swapped for another trigger component

File: site/docs/components/file-drop-zone/examples.mdx (hunk 1 above)
Comment (Fercas123) [reply site/docs/components/file-drop-zone/examples.mdx]: sorry, a bit confused... @mikearildbrown `File drop zone` or `Default`?

File: site/src/examples/file-drop-zone/Error.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  FileDropZoneIcon,
+  FileDropZoneTrigger,
+} from "@salt-ds/lab";
+
+export const Error = (): ReactElement => (
+  <FileDropZone style={{ width: 200 }} status="error">
+    <FileDropZoneIcon status="error" />
+    <strong>Error uploading</strong>
+    <FileDropZoneTrigger accept=".png" />
+    <p>Only .png files</p>
```
[the remark this answers is not in the document]
Comment (ivan-calderon) [reply site/src/examples/file-drop-zone/Error.tsx:13]: Please also check spacing between elements into the FDZ. It seems there is a big gap between the button and the validation criteria text.

File: site/src/examples/file-drop-zone/Disabled.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  FileDropZoneTrigger,
+} from "@salt-ds/lab";
+import { Text } from "@salt-ds/core";
+
+export const Disabled = (): ReactElement => (
+  <FileDropZone style={{ width: 300 }} disabled>
+    <FileDropZoneIcon />
+    <strong>Drop files here or</strong>
+    <FileDropZoneTrigger disabled />
+    <Text>Only .png files</Text>
```
Comment (ivan-calderon) [inline site/src/examples/file-drop-zone/Disabled.tsx:14]: Validation criteria text is not inheriting disabled state styles.

File: site/src/examples/file-drop-zone/Error.tsx [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+import {
+  FileDropZone,
+  FileDropZoneIcon,
+  FileDropZoneTrigger,
+} from "@salt-ds/lab";
+
+export const Error = (): ReactElement => (
+  <FileDropZone style={{ width: 300 }} status="error">
+    <FileDropZoneIcon status="error" />
+    <strong>File format is not allowed</strong>
```
Comment (ivan-calderon) [inline site/src/examples/file-drop-zone/Error.tsx:12]: Fer, I changed the title text to "File format is not allowed" here, to align the example to our recommendations. Hope that's ok!

Comment (Fercas123) [reply site/src/examples/file-drop-zone/Error.tsx:12]: thanks! :)

**Review by ivan-calderon — APPROVED** [review]
